eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,173 @@
+    {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+function Test-PreviousScript() 
+{
+  $result = $null
+  if($LASTEXITCODE -eq 0)
+  {
+    $result = "succeeded"
+  }
+  else 
+  {
+    $result = "failed"
+  }
+  Write-Output $result
+}
+
+$sdkPaths = $sdkPaths | select -Unique
+$packages = @()
+$artifactsPath = "$RepoRoot/artifacts/packages"
+if(!(Test-Path $artifactsPath))
+{
+  New-Item -ItemType Directory -Force -Path $artifactsPath
+}
+foreach ($sdkPath in $sdkPaths)
+{
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
+  $result = Test-PreviousScript
+  if($result -eq "succeeded")
+  {
+    Write-Host "Successfully generated code for" $packageName "`n"
+    
+    $csprojPath = Get-ChildItem $srcPath -Filter *.csproj -Recurse
+    dotnet pack $csprojPath --output $artifactsPath /p:RunApiCompat=$false
+    $result = Test-PreviousScript
+    if($result -eq "succeeded")
+    {
+      $artifacts +=  Get-ChildItem $artifactsPath -Filter *$packageName* -Recurse | Select-Object -ExpandProperty FullName | Resolve-Path -Relative
+
+      $logFilePath = Join-Path "$srcPath" 'log.txt'
+      if(!(Test-Path $logFilePath))
+      {
+        New-Item $logFilePath
+      }
+      dotnet build $csprojPath /t:RunApiCompat /p:TargetFramework=netstandard2.0 /flp:v=m`;LogFile=$logFilePath
+      
+      $result = Test-PreviousScript
+      if($result -eq "succeeded")
+      {
+        $hasBreakingChange = $false
+      }
+      else
+      {
+        $logFile = Get-Content -Path $logFilePath | select-object -skip 2
+        $breakingChanges = foreach($Obj in $logFile) 
+        {       
+          $begin = ""
+          $end = ",`n"
+          $begin + $Obj + $end
+        }
+        $content = "Breaking Changes: $breakingChanges"
+        $hasBreakingChange = $true
+        $result = "succeeded"
+      }
+      if (Test-Path $logFilePath) 
+      {
+        Remove-Item $logFilePath
+      }
+    }
+  } 
+  else 
+  {
+    Write-Host "Error occurred while generating code for" $packageName "`n"
+  }
+
+  $path = @()
+  $path += $sdkPath
+
+  $packageNameArr = $packageName.Split(".")
+  $name = $input.relatedReadmeMdFiles -match $packageNameArr[2]
+  $readmeMd = @()
+  if($packageName -match "Azure.ResourceManager")
+  {
+    $readmeMd += $name -match "resource-plane"
+  }
+  elseif($packageName -match "Azure.")
+  {
+    $readmeMd += $name -match "data-plane"
+  }
+
+  $changelog = [PSCustomObject]@{
+    content = $content
+    hasBreakingChange = $hasBreakingChange
+    }
+
+  $downloadUrlPrefix = $input.installInstructionInput.downloadUrlPrefix
+  $installInstructions = [PSCustomObject]@{
+    full = "Download the $packageName from [here]($downloadUrlPrefix)"
```

Comment:
I'm not sure that link will work so we should test to see what it does. I was expecting we would need to add the package name to the download line as well to get this link to be useful.

Reply:
Yes it should be prefix + filename

Reply:
Updated!